### ISSUE TITLE
Cannot handle error incase connection is interupted

### ISSUE THREAD

**Issue by duongvansang** [gdaws/node-stomp#85]
1. Start ActiveMQ
2. Connect as normally (connect, subscribe topic)
3. Stop ActiveMQ suddently
4. Throw exception and crash application, client.on('error', [callback]), socket.on('error', [callback]) does not work.
events.js:163
      throw er; // Unhandled 'error' event
      ^
Error: connection ended without disconnect receipt
So cannect run reconnect logic

**Comment by duongvansang** [comment]
No, actually connection is interrupted unexpected, like server die, router or internet line broken down.
To produce is very simple
1. You start activeMQ server on your local
2. Run a node app and subscribed a topic
3. Continue to send message to topic
4. Stop ActiveMQ 
5. You will see error log and application is crashed. No way to handle error and try to reconnect
As I debug, It will go to else branch of this method.

Client.prototype._onInputEnd = function() {
  if (this._hasDisconnectReceipt) {
    this.destroy();
  }
  else {
    
    var errorMessage = this.hasFinishedOutput() ? 
      'connection ended without disconnect receipt' : 
      'connection ended unexpectedly';
    
    this.destroy(this.createProtocolError(errorMessage));
  }
};

**Comment by gdaws** [comment]
I'm running Node 10.8.0 and following your steps but can't reproduce the application crash. Adding an error listener to the client in connect callback avoids the error being sent to the uncaught exception handler.

I notice that the error message is affected by the halfOpen connect setting. The error message is 'connection ended without disconnect receipt' when halfOpen value is false (default value) and 'connection ended unexpectedly' when halfOpen value is true.

**Comment by duongvansang** [comment]
Little strange, could you share your test code?

**Comment by gdaws** [comment]
Here's my test code:

```js
stompit.connect(server, function(error, client) {

    if (error) {
        console.error(error);
        return;
    }

    client.on('error', function(error) {
        console.error(error);
    });

    setInterval(function() {

        const headers = {
            destination: '/queue/test',
            'content-type': 'text/plain'
        };

        const frame = client.send(headers);

        frame.write('hello');
        frame.end();

        console.log('send');

    }, 1000);

    console.log('connected');
});
```
